DOC document that last step is never cached in pipeline

windiana42 opened the pull request:
In compose.rst and pipeline.py there are three places where pipeline caching is explained. An extra sentence was added that currently, the last step will never be cached. In one place it is mentioned that this might change in the future.

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs

Improves #23112. This just improves documentation regarding the current state. There might be more activity on the Issue in order to change the behavior of scikit-learn to extend caching functionality. 
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

In `compose.rst` and `pipeline.py`, there are three places where pipeline caching is explained. An extra sentence was added explaining that currently, the last step will never be cached. In one place it is mentioned that this might change in the future.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

windiana42 committed:
#23112: fix documentation last step is not cached

In compose.rst and pipeline.py there are three places where pipeline caching is explained. An extra sentence was added that currently, the last step will never be cached. In one place it is mentioned that this might change in the future.

windiana42 committed:
Merge remote-tracking branch 'upstream/main' into fix_doc_23112

windiana42 commented:
@betatim

betatim reviewed (approved):
Looks good to me

betatim commented:
Maybe @glemaitre has a second to provide a second review/if this is what he was thinking of in his comment on the issue.

glemaitre commented on doc/modules/compose.rst:
```diff
@@ -198,7 +198,8 @@ after calling ``fit``.
 if the parameters and input data are identical. A typical example is the case of
 a grid search in which the transformers can be fitted only once and reused for
-each configuration.
+each configuration. Currently, the last step will never be cached. This might
+change in the future.
```
I don't think that we need to mention this.

```suggestion
each configuration. Currently, the last step will never be cached.
```

glemaitre commented on sklearn/pipeline.py:
```diff
@@ -80,11 +80,11 @@ class Pipeline(_BaseComposition):
-        the caching directory. Enabling caching triggers a clone of
-        the transformers before fitting. Therefore, the transformer
-        instance given to the pipeline cannot be inspected
+        Used to cache the fitted transformers of the pipeline. Currently,
+        the last step will never be cached. By default, no caching is performed.
```
```suggestion
        Used to cache the fitted transformers of the pipeline. The last step will never be cached, even if it is a transformer. By default, no caching is performed.
```

glemaitre replied:
You can apply the same changes in below.

windiana42 replied:
@glemaitre I think I implemented your suggestion consistently across the three touched locations, now.

windiana42 committed:
Update doc/modules/compose.rst

No reference to future changes needed.

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

windiana42 committed:
suggestion: emphasize on last step transformer

Implement suggestion to emphasize that last step is not cached even if it is a transformer.

glemaitre reviewed (approved):
LGTM. Enabling the auto-merge. Thanks @windiana42
